pkg/types/network.go

```diff
@@ -110,10 +110,10 @@ type Listener interface {
 	SetPerConnBufferLimitBytes(limitBytes uint32)
 
 	// Set if listener should use original dst
-	SetUseOriginalDst(use bool)
+	SetUseOriginalDst(use v2.OriginalDstType)
```

Comment:
SetUseOriginalDst rename to SetOriginalDst and  add a new interface named GetOriginalDst

Reply:
use rename to type